src/batch/src/monitor/stats.rs

```diff
@@ -0,0 +1,418 @@
+                "batch_task_fast_poll_duration",
+                "The total duration (s) of fast polls.",
+            ),
+            &task_labels[..],
+        )
+        .unwrap();
+        descs.extend(task_fast_poll_duration.desc().into_iter().cloned());
+
+        let task_idle_duration = GaugeVec::new(
+            opts!(
+                "batch_task_idle_duration",
+                "The total duration (s) that tasks idled.",
+            ),
+            &task_labels[..],
+        )
+        .unwrap();
+        descs.extend(task_idle_duration.desc().into_iter().cloned());
+
+        let task_poll_duration = GaugeVec::new(
+            opts!(
+                "batch_task_poll_duration",
+                "The total duration (s) elapsed during polls.",
+            ),
+            &task_labels[..],
+        )
+        .unwrap();
+        descs.extend(task_poll_duration.desc().into_iter().cloned());
+
+        let task_scheduled_duration = GaugeVec::new(
+            opts!(
+                "batch_task_scheduled_duration",
+                "The total duration (s) that tasks spent waiting to be polled after awakening.",
+            ),
+            &task_labels[..],
+        )
+        .unwrap();
+        descs.extend(task_scheduled_duration.desc().into_iter().cloned());
+
+        let task_slow_poll_duration = GaugeVec::new(
+            opts!(
+                "batch_task_slow_poll_duration",
+                "The total duration (s) of slow polls.",
+            ),
+            &task_labels[..],
+        )
+        .unwrap();
+        descs.extend(task_slow_poll_duration.desc().into_iter().cloned());
+
+        let metrics = Self {
+            descs,
+            task_first_poll_delay,
+            task_fast_poll_duration,
+            task_idle_duration,
+            task_poll_duration,
+            task_scheduled_duration,
+            task_slow_poll_duration,
+            delete_task: Arc::new(Mutex::new(Vec::new())),
+        };
+        registry.register(Box::new(metrics.clone())).unwrap();
+        metrics
+    }
+
+    /// Create a new `BatchTaskMetrics` instance used in tests or other places.
+    pub fn for_test() -> Self {
+        Self::new(prometheus::Registry::new())
+    }
+
+    fn clean_metrics(&self) {
+        let delete_task: Vec<TaskId> = if let Ok(mut delete_task) = self.delete_task.try_lock() {
+            if delete_task.is_empty() {
+                return;
+            }
+            std::mem::take(delete_task.as_mut())
+        } else {
+            return;
+        };
+        for id in &delete_task {
+            let stage_id = id.stage_id.to_string();
+            let task_id = id.task_id.to_string();
+            let labels = vec![id.query_id.as_str(), stage_id.as_str(), task_id.as_str()];
+
+            macro_rules! remove {
+                        ($({ $metric:ident, $type:ty},)*) => {
+                            $(
+                                self.$metric.remove_label_values(&labels).unwrap();
```

Comment:
> If in the same executor, we call create_collector_for_xxx with the same set of labels twice.

Would this case happen normally?🤔 Unwrap() here based on the assumption that before executor execute, it will create a collector. The same executor willn't create a collector twice. For now the executor metrics follow this assumption, but I am not sure whether we should assert it alway true. (Seems it's reasonable for me).

Reply:
Then maybe some comment about this assumption?